src/lib/fetcher.ts

```diff
@@ -0,0 +1,81 @@
+import IntentList from './intent-list'
+import APIContentType from '../lib/interfaces/content-type'
+import { ContentType } from '../lib/entities/content-type'
+import * as _ from 'lodash'
+import * as Bluebird from 'bluebird'
+
+export default class Fetcher {
+  private makeRequest: any
+
+  constructor (makeRequest) {
+    this.makeRequest = makeRequest
+  }
+
+  async getEntriesInIntents (intentList: IntentList) {
+    const ids: Array<string> = _.uniq(intentList.getIntents().map((intent) => intent.getContentTypeId()))
```

Comment:
I don't think it makes a huge difference but I prefer `const ids: string[]` instead of `Array<string>`